code/game/machinery/doors/door.dm

```diff
@@ -77,9 +77,14 @@
 		var/mob/living/M = AM
 		if(world.time - M.last_bumped <= 10) return	//Can bump-open one airlock per second. This is to prevent shock spam.
 		M.last_bumped = world.time
-		if(!M.restrained() && M.mob_size > MOB_SIZE_SMALL)
-			bumpopen(M)
-		return
+		if(!M.restrained())
+			if(istype(AM, /mob/living/simple_animal))
+				var/mob/living/simple_animal/A = AM
+				if(A.can_collar)

```

Comment:
Revising this to include if they are both collarable, and are wearing a collar. (Though.. I suppose just checking if the are wearing a collar would do.. maybe rather check if they are wearing a collar with an ID?)